Microsoft.HealthVault.Fhir/Transformers/ImmunizationToHealthVault.cs

```diff
@@ -0,0 +1,122 @@
+            }
+            hvImmunization.AnatomicSurface = fhirImmunization.Site.ToCodableValue();
+
+            Extension immunizationExtension = fhirImmunization.GetExtension(HealthVaultExtensions.ImmunizationDetail);
+            if (immunizationExtension != null)
+            {
+                hvImmunization.Sequence = immunizationExtension.GetStringExtension(HealthVaultExtensions.ImmunizationDetailSequence);
+                hvImmunization.Consent = immunizationExtension.GetStringExtension(HealthVaultExtensions.ImmunizationDetailConcent);
+                hvImmunization.AdverseEvent = immunizationExtension.GetStringExtension(HealthVaultExtensions.ImmunizationDetailAdverseEvent);
+            }
+
+            if (!fhirImmunization.Note.IsNullOrEmpty())
+            {
+                fhirImmunization.Note.ForEach(note =>
+                {
+                    string separator = string.Empty;
+                    if (!string.IsNullOrEmpty(hvImmunization.CommonData.Note))
+                    {
+                        separator = Environment.NewLine; //Let's seperate each note with new line
+                    }
+
+                    hvImmunization.CommonData.Note += $"{note.Text}";
+                });
+            }
+
+            return hvImmunization;
+        }
+
+        private static CodableValue GetManufacturer(Hl7.Fhir.Model.Immunization fhirImmunization)
+        {
+            var manufacturerComponent = fhirImmunization.Manufacturer;
+            if (manufacturerComponent.IsContainedReference)
+            {
+                var containedReference = fhirImmunization.Contained.SingleOrDefault(resouce =>
+                    resouce.Id.Equals(manufacturerComponent.Reference) && resouce.GetType().Equals(typeof(Hl7.Fhir.Model.Organization)));
+
+                if (containedReference == null)
+                {
+                    return null;
+                }
+
+                return new CodableValue((containedReference as Hl7.Fhir.Model.Organization).Name);
+            }
+
+            if (string.IsNullOrEmpty(manufacturerComponent.Display))
+                return null;
+
+            return new CodableValue(manufacturerComponent.Display);
+        }
+
+        private static PersonItem GetProvider(Hl7.Fhir.Model.Immunization fhirImmunization)
+        {
+            var practitionerComponent = fhirImmunization.Practitioner.First(); //Let's take only the first one
+            if (practitionerComponent.Actor.IsContainedReference)
+            {
+                var containedReference = fhirImmunization.Contained.SingleOrDefault(resouce =>
+                    resouce.Id.Equals(practitionerComponent.Actor.Reference) && resouce.GetType().Equals(typeof(Practitioner)));
+
+                if (containedReference == null)
+                {
+                    return null;
+                }
+
+                return (containedReference as Practitioner).ToHealthVault();
```

Comment:
>(containedReference as Practitioner [](start = 23, length = 35)

This could throw an exception, why not do the cast before checking for null?

Reply:
It is done in the lambda expression to find containedReference

var containedReference = fhirImmunization.Contained.SingleOrDefault(resouce =>
                    resouce.Id.Equals(practitionerComponent.Actor.Reference) && resouce.GetType().Equals(typeof(Practitioner)));